src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -840,8 +854,32 @@ private void initChannelPipeline(ChannelPipeline pipeline) {
             pipeline.addLast("outboundGlobalStateHandler", new OutboundGlobalStateHandler(this));
         }
 
-        pipeline.addLast("handler", this);
+        if(separateProcessingEventLoop) {
```

Comment:
allow to run main handler of client to proxy connection in separate thread. So actually it allow to decouple read and request handling.